src/praisonai-agents/praisonaiagents/snapshot/snapshot.py

```
@@ -238,6 +238,22 @@ def _sync_files(self):
                 except (IOError, OSError) as e:
                     logger.warning(f"Failed to copy {src_path}: {e}")
         
+        # Prune shadow files whose source has been removed from the project,
+        # so a later commit can record the deletion (the shadow tree must
+        # mirror the project, not be append-only).
+        tracked_now = set(files_to_track)
+        for root, dirs, files in os.walk(self.shadow_path):
```

Ruff (0.16.0)

[warning] Loop control variable `dirs` not used within loop body

(B007)